concent_api/api-integration-test.py

```diff
@@ -0,0 +1,208 @@
+import random
+import time
+import http.client
+from base64                 import b64encode
+
+from golem_messages         import dump
+from golem_messages         import load
+from golem_messages.message import AckReportComputedTask
+from golem_messages.message import ComputeTaskDef
+from golem_messages.message import ForceReportComputedTask
+from golem_messages.message import Message
+from golem_messages.message import TaskToCompute
+
+from utils.testing_helpers  import generate_ecc_key_pair
+
+import requests
+
+CONCENT_PUBLIC_KEY  = b'D\x1e\x9d\xfe\x0b3\x1a\x9e\xaf\xe8\x1d^\xbb\xf1~\xb1\x15A\xce\xda\xf5\x8f\xe4\x166\x9b\xa9\x00\xe1+]\xcd\xe7\x7f\x93\x17t\xa6\xbd\xbfY\xcf|]\xa5\x9e;|lJ\xbf(q\xb7`\x12S\xcd\n5\xdey\x86\xa5'
+CONCENT_PRIVATE_KEY = b'[\x10!\xca\rv\xb1\x8d7\x84\xd2\x83\xb1\xa5\xab\xf2\x1c\xe0\xb0\x04Pj>hE|tr~,^\xb5'
+
+(PROVIDER_PRIVATE_KEY,  PROVIDER_PUBLIC_KEY)  = generate_ecc_key_pair()
+(REQUESTOR_PRIVATE_KEY, REQUESTOR_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+def print_golem_message(message, indent = 4):
+    assert isinstance(message, Message)
+
+    HEADER_FIELDS  = ['timestamp', 'encrypted', 'sig']
+    PRIVATE_FIELDS = {'_payload', '_raw', 'result_hash'}
+    assert 'type' not in message.__slots__
+    fields = ['type'] + HEADER_FIELDS + sorted(set(message.__slots__) - set(HEADER_FIELDS) - PRIVATE_FIELDS)
+    values = [
+        type(message).__name__ if field == 'type' else
+        getattr(message, field)
+        for field in fields
+    ]
+
+    for field, value in zip(fields, values):
+        if isinstance(value, bytes):
+            try:
+                nested_message = load(
+                    value,
+                    CONCENT_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY,
+                )
+            except AttributeError:
+                # FIXME: golem-messages provides no reliable way to discern invalid messages from other AttributeErrors.
+                print('{}{:30} = <BINARY DATA>'.format(' ' * indent, field))
+            else:
+                print('{}{:30} ='.format(' ' * indent, field))
+                print_golem_message(nested_message, indent = indent + 4)
+        else:
+            print('{}{:30} = {}'.format(' ' * indent, field, value))
+
+
+def api_request(host, endpoint, private_key = None, data = None, headers = None, ):
+    assert all(value not in ['', None] for value in [endpoint, host, headers])
+    import ipdb; ipdb.set_trace()
+    url = "{}/api/v1/{}/".format(host, endpoint)
+
+    if data is None:
+        print('RECEIVE ({})'.format(url))
+    else:
+        print('SEND ({})'.format(url))
+        print('MESSAGE:')
+        print_golem_message(data)
+
+        data = dump(
+            data,
+            private_key,
+            CONCENT_PUBLIC_KEY,
+        )
+
+    if data is None:
+        response = requests.post("{}".format(url), headers = headers)
+    else:
+        response = requests.post("{}".format(url), headers = headers, data = data)
```

Comment:
There is no need to split that to if, you can pass data to requests.post() even if it is None.